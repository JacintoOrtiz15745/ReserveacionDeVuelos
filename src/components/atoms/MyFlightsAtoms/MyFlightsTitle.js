import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {styles} from '../../../styles/MyFlightsStyles';
import {colors} from '../../../styles/Colors';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {connect} from 'react-redux';
import {signoutFunction} from '../../../../redux/actions/AuthActions';

const MyFlightsTitle = ({title, signoutFunction}) => {
  return (
    <View style={styles.titleContainer} testID="view">
      <Text style={styles.myFlightsTitle} testID="text">
        {title}
      </Text>
      <TouchableOpacity onPress={signoutFunction}>
        <Icon name="logout" size={35} color={colors.Purple} />
      </TouchableOpacity>
    </View>
  );
};

const mapDispatchToProps = {signoutFunction};

export default connect(null, mapDispatchToProps)(MyFlightsTitle);
