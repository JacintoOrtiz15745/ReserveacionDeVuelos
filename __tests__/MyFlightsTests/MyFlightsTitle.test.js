import React from 'react';
import {shallow} from 'enzyme';
import MyFlightsTitle from '../../src/components/atoms/MyFlightsAtoms/MyFlightsTitle';
import {connect} from 'react-redux';
import configureMockStore from 'redux-mock-store';

const props = {
  title: 'title',
};

describe('Render correctly', () => {
  const mockStore = configureMockStore([]);
  let store = mockStore({});
  const ConnectedComponent = connect()(MyFlightsTitle);
  const wrapper = shallow(<ConnectedComponent {...props} store={store} />);
  const component = wrapper.dive().dive();
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Render correctly', () => {
    const render = component.findWhere(
      (node) => node.prop('testID') === 'view',
    );
    expect(render.exists()).toBeTruthy();
  });
  test('Title correctly', () => {
    const title = component.findWhere((node) => node.prop('testID') === 'text');
    expect(title.props().children).toEqual(props.title);
  });

});
