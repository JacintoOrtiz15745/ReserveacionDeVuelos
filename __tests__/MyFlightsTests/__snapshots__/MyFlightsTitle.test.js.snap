// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly snapshot 1`] = `
<Connect(MyFlightsTitle)
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  title="title"
/>
`;
